polars/polars-io/src/csv/buffer.rs

```diff
@@ -98,7 +99,7 @@ where
                         return self.parse_bytes(
                             bytes,
                             ignore_errors,
-                            needs_escaping,
+                            needs_escaping && is_whitespace(first_byte), // we only need to do escaping if the first byte was not whitespace
```

Comment:
we already know that first byte is whitespace due to line 97 and we just skipped whitespace, so we can set `needs_escaping` to false

Reply:
`first_byte` will hold the first byte before it's discarded from the array, so it won't necessarily be the same as `bytes[0]`. The case where this matters is something like `  "  4.1"` where we go `  "  4.1"` -> `"  4.1"` ->`4.1"` so we'd need to do the escaping on the first retry however I could not craft a test case where  `  "  4.1"` gets passed to this parser (it is recognized as Utf8 instead of Float64).